src/agent/onefuzz/src/syncdir.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{
+    az_copy,
+    blob::BlobContainerUrl,
+    jitter::jitter,
+    monitor::DirectoryMonitor,
+    telemetry::{Event, EventData},
+    uploader::BlobUploader,
+};
+use anyhow::Result;
+use futures::stream::StreamExt;
+use std::{path::PathBuf, str, time::Duration};
+use tokio::fs;
+
+#[derive(Debug, Clone, Copy)]
+pub enum SyncOperation {
+    Push,
+    Pull,
+}
+
+const DELAY: Duration = Duration::from_secs(10);
+const DEFAULT_CONTINUOUS_SYNC_DELAY: Duration = Duration::from_secs(60);
+
+#[derive(Debug, Deserialize, Clone, PartialEq)]
+pub struct SyncedDir {
+    pub path: PathBuf,
+    pub url: BlobContainerUrl,
+}
+
+impl SyncedDir {
+    pub async fn sync(&self, operation: SyncOperation) -> Result<()> {
+        let dir = &self.path;
+        let url = self.url.url();
+        let url = url.as_ref();
+        verbose!("syncing {:?} {}", operation, dir.display());
+        match operation {
+            SyncOperation::Push => az_copy::sync(dir, url).await,
+            SyncOperation::Pull => az_copy::sync(url, dir).await,
+        }
+    }
+
+    pub async fn init_pull(&self) -> Result<()> {
+        self.init().await?;
+        self.sync(SyncOperation::Pull).await
+    }
+
+    pub async fn init(&self) -> Result<()> {
```

Comment:
Can we add doc comments to these new methods that describe their assumptions and intended behavior?